### PR TITLE
Task 103791 & 103793: Edit Declined Journey  & Error handling

### PR DESCRIPTION
**Pull request by Lisa-V-Sau**
- Declined journey
- Error handling (NOTE error messages are still to be updated on the frontend)
- Renamed file for Edit Approved journey

### CODE REVIEW

File: ApplyToBecomeInternal/ApplyToBecomeCypressTests/cypress/integration/Record_Decision/103787_error-handling.js [hunk 1]
```diff
@@ -31,4 +31,21 @@ describe('103787 Error handling', () => {
...
+        cy.get('[id="AdvisoryBoardDecision-error-link "]').should('contain.text', 'Please select the result of the decision')
+        cy.get('[id="declined-radio"]').click()
+        cy.continueBtn().click()
+        // Who made this decision:
+        cy.continueBtn().click()
+        cy.get('[id="DecisionMadeBy-error-link "]').should('contain.text', 'Please select who made the decision')
+        cy.get('[id="regionaldirectorforregion-radio"]').click()
+        cy.continueBtn().click()
+        // Why was the project declined?
+        // **Error Message needs updating on Frontend**
```
Comment (jordan-beard) [inline ApplyToBecomeInternal/ApplyToBecomeCypressTests/cypress/integration/Record_Decision/103787_error-handling.js:48]: Is this meant to be commented out?

File: ApplyToBecomeInternal/ApplyToBecomeCypressTests/cypress/integration/Record_Decision/103791_edit_declined-decision.js [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+         //select iniital decision
+        cy.get('[id="declined-radio"]').click()
+        
+        // clicks on the continue button
+        cy.continueBtn().click()
+        // selects regional director button
+        cy.get('[id="otherregionaldirector-radio"]').click()
+        // clicks on the continue button
+        cy.continueBtn().click()
+        cy.get('[id="declined-reasons-performance"]')
```
Comment (jordan-beard) [inline ApplyToBecomeInternal/ApplyToBecomeCypressTests/cypress/integration/Record_Decision/103791_edit_declined-decision.js:91]: Is it worth moving some of this duplicated code into its own function?

**Comment by jordan-beard** [comment]
Should there be a test to check that the summary displayed on the task list page is displaying the correct data?

File: ApplyToBecomeInternal/ApplyToBecomeCypressTests/cypress/integration/Record_Decision/103787_error-handling.js (hunk 1 above)
Comment (Lisa-V-Sau) [reply ApplyToBecomeInternal/ApplyToBecomeCypressTests/cypress/integration/Record_Decision/103787_error-handling.js:48]: Error messages on frontend are not yet updated so this is the place holder

**Review by MartinWheelerMT — COMMENTED** [review]
As Jordan has mention if it's possible to move some of the shared code into own function, that would be great

**Comment by Lisa-V-Sau** [comment]
> Should there be a test to check that the summary displayed on the task list page is displaying the correct data?

Yes I intend to but I have to access the database and deleted the records, as it is unstable at its current state.

**Review by tim-wilde-nimble — APPROVED** [review]

**Review by MartinWheelerMT — APPROVED** [review]

**Review by cshnimble — APPROVED** [review]

**Pull request merged**